### ISSUE TITLE
[BUG] Message Trees/Message Templates do not parse HL7 v2.x messages in Filter Editor or Transform Editor in Mirth 4.3

### ISSUE THREAD

**Issue by DevinSanders** [nextgenhealthcare/connect#5738]
Recently updated one Mirth instance from 3.8 to 4.2. A second Mirth instance was a clean install of 3.11 then updated to 4.2. 

 - OS: Windows Server 2016 and Windows Server 2019
 - Java Distribution/Version OpenJDK20
 - Connect Version 4.2

1. Edit any channel
2. View either the Source or Destination, the issue occurs in both
3. Edit the Filter or Transformer, the issue occurs in both
4. Paste an HL7 message into the Message Template
5. Switch back to the Message Tree

Expected Behavior
Message tree should display each segment and field of the HL7 

Actual Behavior
The Message Tree never loads. It continues to say "Enter a message template," or if you close the channel editor and come back to a previously edited filter or transformer sometimes the message tree is just blank. The progress bar in the bottom right says "parsing" and never completes. Mirth log file only seems to report errors in deployed channels so there are no error messages related to this. 

Modifying the HL7 version in MSH-12 is ineffective. This seems to affect any HL7 v2.x

Sample Message: 
MSH|^~\&|APP1|FAC1|APP2|FAC2|20221012131737||ADT^A02|c40a177af31e4d82a24e2ba3bf8e4301|P|2.3.1|||NE|NE
PID||P1234|4321P||LASTNAME^FIRSTNAME^MIDDLENAME||20010220|M||MA
PV1||I|^PHOENIX^INFIRMARY^1001^D^153|||^PHOENIX^RECEPTION^1||||||||||||||||||||||||||||||||||||||20221012111737

![Snag_183d7c8d](https://user-images.githubusercontent.com/33848976/228584175-fe4b6786-a9c3-4c02-90b0-74e3b9970c85.png)
![Snag_183d9313](https://user-images.githubusercontent.com/33848976/228584209-f9d08c76-0445-4bb1-8615-5d36084c145a.png)
![Snag_183dab0f](https://user-images.githubusercontent.com/33848976/228584238-c94dfea4-633c-4bc0-9888-9edba04a8cd0.png)
![Snag_183dc28f](https://user-images.githubusercontent.com/33848976/228584273-19216749-c2cb-4719-85db-6686b343f853.png)

**Comment by DevinSanders** [comment]
Enabling the Java Console for Mirth Connect Administrator shows this exception when parsing a message template

```Exception in thread "Timer-13" java.lang.NoSuchMethodError: 'ca.uhn.hl7v2.validation.ValidationContext ca.uhn.hl7v2.validation.impl.ValidationContextFactory.defaultValidation()'
	at ca.uhn.hl7v2.DefaultHapiContext.<init>(DefaultHapiContext.java:100)
	at ca.uhn.hl7v2.DefaultHapiContext.<init>(DefaultHapiContext.java:91)
	at com.mirth.connect.plugins.datatypes.hl7v2.ER7Serializer.<init>(ER7Serializer.java:70)
	at com.mirth.connect.plugins.datatypes.hl7v2.HL7v2DataTypeDelegate.getSerializer(HL7v2DataTypeDelegate.java:27)
	at com.mirth.connect.plugins.DataTypeClientPlugin.getSerializer(DataTypeClientPlugin.java:33)
	at com.mirth.connect.client.ui.TreePanel.setMessage(TreePanel.java:429)
	at com.mirth.connect.client.ui.TemplatePanel$1UpdateTimer.run(TemplatePanel.java:201)
	at java.base/java.util.TimerThread.mainLoop(Timer.java:566)
	at java.base/java.util.TimerThread.run(Timer.java:516)```

**Comment by pacmano1** [comment]
And you have added all the required mcserver/mcservice vmoptions for java 9+, correct?

Can you backup your java version a bit on the server if you have added the required options?

I assume you are using the mirth admin laucher 1.3.1 and the defaul JRE in that for the admin tool, correct?

**Comment by DevinSanders** [comment]
Correct, here is my vmoptions:

```-server
-Xmx4096m
-Djava.awt.headless=true
-Dapple.awt.UIElement=true
--add-modules=java.sql.rowset
--add-exports=java.base/com.sun.crypto.provider=ALL-UNNAMED
--add-exports=java.base/sun.security.provider=ALL-UNNAMED
--add-opens=java.base/java.lang=ALL-UNNAMED
--add-opens=java.base/java.lang.reflect=ALL-UNNAMED
--add-opens=java.base/java.math=ALL-UNNAMED
--add-opens=java.base/java.net=ALL-UNNAMED
--add-opens=java.base/java.security=ALL-UNNAMED
--add-opens=java.base/java.security.cert=ALL-UNNAMED
--add-opens=java.base/java.text=ALL-UNNAMED
--add-opens=java.base/java.util=ALL-UNNAMED
--add-opens=java.base/sun.security.pkcs=ALL-UNNAMED
--add-opens=java.base/sun.security.rsa=ALL-UNNAMED
--add-opens=java.base/sun.security.x509=ALL-UNNAMED
--add-opens=java.desktop/java.awt=ALL-UNNAMED
--add-opens=java.desktop/java.awt.color=ALL-UNNAMED
--add-opens=java.desktop/java.awt.font=ALL-UNNAMED
--add-opens=java.xml/com.sun.org.apache.xalan.internal.xsltc.trax=ALL-UNNAMED```

**Comment by DevinSanders** [comment]
Administrator Launcher 1.3.1. I attempted with both the Bundled and the Default Java Home. I have tried running Administrator from a remote computer and from the local server with Mirth Connect Server installation. 

![Snag_185308aa](https://user-images.githubusercontent.com/33848976/228590883-0fc6befd-0744-42b6-b286-9eb908017805.png)

**Comment by pacmano1** [comment]
And backing down the java version on the server?  e.g. like v11.   Not that latest should not work, but good to report on it.

**Comment by DevinSanders** [comment]
In Mirth 3.8 this functionality worked on OpenJDK 17, so I tried going back to that version. Same result, and same exception in the Java Console from Mirth Administrator:

```Using JRE version 17.0.2+8-86 OpenJDK 64-Bit Server VM
Java home directory = c:\program files\openjdk\jdk-17.0.2
User home directory = C:\Users\devin.sanders
----------------------------------------------------
l:  Clear console window
s:  Dump system properties
----------------------------------------------------
Mar 29, 2023 10:44:25 AM com.sun.javafx.application.PlatformImpl startup
WARNING: Unsupported JavaFX configuration: classes were loaded from 'unnamed module @15aab8c6'
Exception in thread "Timer-4" java.lang.NoSuchMethodError: 'ca.uhn.hl7v2.validation.ValidationContext ca.uhn.hl7v2.validation.impl.ValidationContextFactory.defaultValidation()'
	at ca.uhn.hl7v2.DefaultHapiContext.<init>(DefaultHapiContext.java:100)
	at ca.uhn.hl7v2.DefaultHapiContext.<init>(DefaultHapiContext.java:91)
	at com.mirth.connect.plugins.datatypes.hl7v2.ER7Serializer.<init>(ER7Serializer.java:70)
	at com.mirth.connect.plugins.datatypes.hl7v2.HL7v2DataTypeDelegate.getSerializer(HL7v2DataTypeDelegate.java:27)
	at com.mirth.connect.plugins.DataTypeClientPlugin.getSerializer(DataTypeClientPlugin.java:33)
	at com.mirth.connect.client.ui.TreePanel.setMessage(TreePanel.java:429)
	at com.mirth.connect.client.ui.TemplatePanel$1UpdateTimer.run(TemplatePanel.java:201)
	at java.base/java.util.TimerThread.mainLoop(Timer.java:566)
	at java.base/java.util.TimerThread.run(Timer.java:516)```

**Comment by pacmano1** [comment]
That seems like the upgrade messed up.    I would try a clean install (no upgrade) and see if the same problem occurs.

**Comment by DevinSanders** [comment]
I can try that. But the last time I tried a clean install of Mirth 4 ~~(I think it was v4.1)~~ (It was version 4.0.1) the database schema did not get applied and Mirth Connect Server was never able to start. The only success I have had with Mirth 4.x is by upgrading from a previous Mirth. On that note, the upgrade only seems to work in small intervals. e.g. when trying to update from Mirth 3.8 to 4.2 I received an error that the schema could not be updated which resulted in a corrupted database. I had to update 3.8 -> 3.11 -> 4.0 -> 4.2 to get a running instance of Mirth Connect 4.2.

**Comment by pacmano1** [comment]
I have updated lots of instances without a problem and clean install all the time, that's very puzzling.   I don't use windows often, however on test VMs I set up for things like your post to help folks  (e.g. win2019 against MSSQL, etc.) it all works fine.  The times DB tables not creating (ince the DB already needs to exist and owned by the mirth jdbc user) not working are always related to bad permissions on the SQL DB itself assuming connectivitty is fine.

**Comment by DevinSanders** [comment]
I always give the Mirth sql login the dbowner permission so that tables can be added or modified without issues. I usually do not have a problem updating by one or two minor versions. 3.8 to 3.12 also failed to apply schema changes, but based on the upgrade notes for 3.12 I expected issues for such a major change. I think there may just be conflicts between much older schemas and the schema changes need to be applied in incremental patches, I have never believed this warranted a bug report.

**Comment by DevinSanders** [comment]
I uninstalled Mirth Connect and confirmed that C:\Program Files\Mirth Connect was completely removed by the installer. I then performed a clean installation of Mirth 4.2. I had no issues with the database since the schema had already been updated from the previous upgrade. The message tree now loads as expected. I will reiterate that I had updated two separate Mirth Connect instances and experienced the same issue. The first instance was updated from 3.8, to 3.11, then to 3.2. The second instance was a clean install of 3.11 then updated to 4.2. This upgrade path was necessary to successfully apply schema changes to a pre-existing SQL database. I was unable to successfully update directly from 3.8 to 4.2 without corrupting the existing database. 

Something is not being properly updated in the Mirth Connect installation when upgrading versions. This is on a Windows Server OS with a MSSQL Server database, I have not tried this with any other environments. It seems that the work-around for this environment is as follows:

1. Backup Mirth database and Mirth Connect configuration files
2. Upgrade Mirth Connect. 
3. When upgrading from a much older version of Mirth Connect, update in small increments of minor versions, applying database schema changes between each version to avoid database corruption. 
4. Uninstall Mirth Connect
5. Clean install Mirth Connect
